source/modes/GameMode.cpp

```diff
@@ -1070,63 +1035,93 @@ bool GameMode::keyPressedChat(const OIS::KeyEvent &arg)
 void GameMode::receiveChat(ChatMessage* message)
 {
     mChatMessages.emplace_back(message);
+
+    // Adds the message right away
+    CEGUI::Window* chatTextBox = getModeManager().getGui().getGuiSheet(Gui::inGameMenu)->getChild("GameChatWindow/GameChatText");
+    chatTextBox->appendText(reinterpret_cast<const CEGUI::utf8*>(message->getMessageAsString().c_str()));
+
+    // Ensure the latest text is shown
+    CEGUI::Scrollbar* scrollBar = reinterpret_cast<CEGUI::Scrollbar*>(chatTextBox->getChild("__auto_vscrollbar__"));
```

Comment:
Are you sure you want to do reinterpret_cast instead of static_cast here? I can see why it's needed for the message text, but don't use reinterpret_cast unless there is a really good reason for it.